### DISCUSSION TITLE
[Feature] Connect to internal ip when connected to known SSID

### DISCUSSION THREAD

**Discussion by HHUBSS** [immich-app/immich#3481] [Feature Request]
### The feature

I don't know if this was already requested before but may I request to have the option of connecting to internal IP when connected to a known SSID, like your home Wi-Fi. This is like in Home Assistant (below image):

![image](https://github.com/immich-app/immich/assets/89816305/030e159a-c962-4bec-8ee4-d409784c8ae0)

In this way, the upload and viewing of photos and videos are much faster. I would like also to request to add the option to upload only when connected to the same SSID.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
Duplicate of #308

**Reply by HHUBSS** [reply]
Ok. I read the thread but what docker container can achieve the DNS split you mention? I tried Adguard but I can't figure it out as I can't input the port on the DNS rewrite rules. Do you have a guide on how to setup the DNS split?

**Reply by radh21301** [reply]
This is how I do it.

For external access:
* I have Nginx Proxy Manager running at a certain IP (port 80). I think you have it running at a different port. I struggled with that as well since I am new to self-hosting. But I gave up and ran NPM at port 80, and it makes things simpler. Note: I run Unraid and it makes it easy running services on br0 with each service on its ow IP.
* I point my external address 'example.domain' to NPM
* NPM then routes it to a certain IP and certain port.

For internal:
* I have my devices using adguard as DNS
* I have adguard running which rewrites 'example.domain' to NPM (since it is on port 80..you dont have to specify any port. Just put the IP address).
* Since NPM is already configured to route 'example.domain' to a certain IP and port..it works internally (within my network)

**Reply by jrasm91** [reply]
This is called split DNS. Public DNS resolves to public IP while local DNS resolves to local IP. For it to be seamless it does need to work with/use the same port both locally and publicly, so 80 or 443 is pretty common, but you could use any.

**Reply by HHUBSS** [reply]
Thank you @radh21301 and @jrasm91. I'll try your suggestions. BRB.

**Comment by Betoniara** [comment]
Why make life difficult for yourself like that?
